firstvoices/backend/models/widget.py

```diff
@@ -0,0 +1,149 @@
+import rules
+from django.db import models
+from django.utils.translation import gettext as _
+
+from backend.models.base import (
+    BaseControlledSiteContentModel,
+    BaseModel,
+    BaseSiteContentModel,
+)
+from backend.permissions import predicates
+
+
+class WidgetTypes(models.IntegerChoices):
+    # enum intentionally has gaps to allow future changes to keep sequential order
+    WIDGET_ALPHABET = 0
+    WIDGET_APPS = 10
+    WIDGET_CONTACT = 20
+    WIDGET_GALLERY = 30
+    WIDGET_IFRAME = 40
+    WIDGET_KEYBOARDS = 50
+    WIDGET_LOGO = 60
+    WIDGET_LIST = 70
+    WIDGET_QUOTES = 80
+    WIDGET_STATS = 90
+    WIDGET_TEXT = 100
+    WIDGET_TEXTCONCISE = 110
+    WIDGET_TEXTFULL = 120
+    WIDGET_TEXTICONS = 130
+    WIDGET_TEXTMULTI = 140
+    WIDGET_WOTD = 150
+
+
+class WidgetFormats(models.IntegerChoices):
+    # enum intentionally has gaps to allow future changes to keep sequential order
+    default = 0
+    left = 10
+    right = 20
+
+
+class Widget(BaseModel):
+    class Meta:
+        verbose_name = _("widget")
+        verbose_name_plural = _("widgets")
+        rules_permissions = {
+            "view": rules.always_allow,
+            "add": predicates.is_superadmin,
+            "change": predicates.is_superadmin,
+            "delete": predicates.is_superadmin,
+        }
+
+    title = models.CharField(max_length=225)
+    type = models.IntegerField(
```

Comment:
Is your IDE complaining that type is a built-in python function name as well? Probably worth changing this to widget_type or something

Reply:
Good catch. Thanks!